[完成翻译] src/content/language/comments.md (#517)
cfug/dart.cn: src/content/language/comments.md

---
title: 注释

> description:  Dart 中不同类型的注释.
> prevpage:
>   url: /language/operators
>   title: 操作符
> nextpage:
>   url: /language/built-in-types
>   title: 基本数据类型

Reviewer: - 建议保留原始元数据

```suggestion
# title: Comments
title: 注释
# description: The different comment types in Dart.
description:  Dart 中不同类型的注释。
prevpage:
  url: /language/operators
  # title: Operators
  title: 操作符
nextpage:
  url: /language/built-in-types
  # title: Built-in types
  title: 基本类型
```

## Multi-line comments

## 多行注释

A multi-line comment begins with `/*` and ends with `*/`. Everything
between `/*` and `*/` is ignored by the Dart compiler (unless the
comment is a documentation comment; see the next section). Multi-line
comments can nest.

> 多行注释以 `/*` 开始，以 `*/` 结束。Dart 编译器会忽略 `/*` 和 `*/` 之间的所有内容（除非该注释是文档注释；请参阅下一节）。多行注释可以嵌套。

Reviewer: - 过长的语句需要换行
- 部分翻译建议

```suggestion
多行注释以 `/*` 开头，以 `*/` 结尾。
Dart 编译器会忽略 `/*` 和 `*/` 之间的所有内容
（除非该注释是文档注释；请参阅下一节）。
多行注释可以嵌套。
```

## Documentation comments

## 文档注释

Documentation comments are multi-line or single-line comments that begin
with `///` or `/**`. Using `///` on consecutive lines has the same
effect as a multi-line doc comment.

> 文档注释是多行或单行注释，以 `///` 或 `/**` 开头。在连续行上使用 `///` 的效果与多行文档注释相同。

Reviewer: - 过长的语句需要换行
- 部分翻译建议

```suggestion
文档注释是以 `///` 或 `/**` 开头的多行或单行注释。
连续多行使用 `///` 的注释效果与多行文档注释等效。
```

Inside a documentation comment, the analyzer ignores all text
unless it is enclosed in brackets. Using brackets, you can refer to
classes, methods, fields, top-level variables, functions, and
parameters. The names in brackets are resolved in the lexical scope of
the documented program element.

> 在文档注释中，分析器会忽略所有未被方括号括起来的文本。通过使用方括号，你可以引用类、方法、字段、顶层变量、函数和参数。方括号中的名称会在被注释程序元素的词法作用域内解析。

Reviewer: - 过长的语句需要换行
- 部分翻译建议

```suggestion
在文档注释中，分析器会处理被方括号括起来的文本内容。
通过使用方括号，你可以引用类、方法、字段、顶层变量、函数和参数。
方括号中的名称将根据注释对象的词法作用域 (Lexical Scope) 进行解析。
```

to the docs for the `feed` method,
and `[Food]` becomes a link to the docs for the `Food` class.

> 在生成的类文档中，`[feed]` 会变成指向 `feed` 方法文档的链接，而 `[Food]` 则会变成指向 `Food` 类文档的链接。

Reviewer: - 过长的语句需要换行
- 部分翻译建议

```suggestion
在生成的类文档中，
`[feed]` 会转换为指向 `feed` 方法文档的链接，
而 `[Food]` 则会转换为指向 `Food` 类文档的链接。
```

To parse Dart code and generate HTML documentation, you can use Dart's
documentation generation tool, [`dart doc`](/tools/dart-doc).
For an example of generated documentation, see the 
[Dart API documentation.]({{site.dart-api}}) 
For advice on how to structure your comments, see
[Effective Dart: Documentation.](/effective-dart/documentation)

> 
> 
> 要解析 Dart 代码并生成 HTML 文档，你可以使用 Dart 的文档生成工具  [`dart doc`](/tools/dart-doc)。有关生成文档的示例，请参阅 [Dart API documentation。]({{site.dart-api}}) 有关如何组织注释的建议，请参阅 [Effective Dart: Documentation。](/effective-dart/documentation)。

Reviewer: - 去除多余的换行
- 过长的语句需要换行
- 部分翻译建议

```suggestion

如果你需要解析 Dart 代码并生成 HTML 文档，
你可以使用 Dart 的文档生成工具 [`dart doc`](/tools/dart-doc)。
关于生成的文档示例，请参阅 [Dart API 文档]({{site.dart-api}})。
关于代码注释规范的建议，请参阅 [高效 Dart 语言指南：文档](/effective-dart/documentation)。
```
